src/backend/distributed/deparser/citus_ruleutils.c

```diff
@@ -390,6 +392,54 @@ pg_get_tableschemadef_string(Oid tableRelationId, IncludeSequenceDefaults
 			}
 #endif
 
+			if (attributeForm->attidentity && includeIdentityDefaults)
+			{
+				Oid seqOid = getIdentitySequence(RelationGetRelid(relation),
+												 attributeForm->attnum, true);
+				Assert(seqOid != InvalidOid);
+
+				char *sequenceName = generate_qualified_relation_name(seqOid);
+
+				if (includeIdentityDefaults == INCLUDE_IDENTITY_AS_SEQUENCE_DEFAULTS)
+				{
+					if (pg_get_sequencedef(seqOid)->seqtypid != INT8OID)
+					{
+						appendStringInfo(&buffer,
+										 " DEFAULT worker_nextval(%s::regclass)",
+										 quote_literal_cstr(sequenceName));
+					}
+					else
+					{
+						appendStringInfo(&buffer, " DEFAULT nextval(%s::regclass)",
+										 quote_literal_cstr(sequenceName));
+					}
+				}
+				else if (includeIdentityDefaults == INCLUDE_IDENTITY)
+				{
+					Form_pg_sequence pgSequenceForm = pg_get_sequencedef(seqOid);
+					uint64 sequenceStart = nextval_internal(seqOid, false);
```

Comment:
Could you explain why we are using this and not simply `pgSequenceForm->seqstart`?

If it has a valid reason, it would be good to add a comment here explaining this.

Reply:
I guess here we are creating a new sequence in `alter_distributed_table` and we want the sequence to start where the old one was left, right?

Reply:
Yes, that's correct. It's also needed for undistribute_table, as well.